packages/flutter/lib/src/cupertino/text_selection_toolbar_button.dart

```diff
@@ -114,25 +110,60 @@ class CupertinoTextSelectionToolbarButton extends StatelessWidget {
   }
 
   @override
-  Widget build(BuildContext context) {
-    final Widget child = this.child ?? Text(
-       text ?? getButtonLabel(context, buttonItem!),
-       overflow: TextOverflow.ellipsis,
-       style: _kToolbarButtonFontStyle.copyWith(
-         color: onPressed != null
-             ? _kToolbarTextColor.resolveFrom(context)
-             : CupertinoColors.inactiveGray,
-       ),
-     );
+  State<StatefulWidget> createState() => _CupertinoTextSelectionToolbarButtonState();
+}
+
+class _CupertinoTextSelectionToolbarButtonState extends State<CupertinoTextSelectionToolbarButton> {
+  bool isPressed = false;
+
+  void _onTapDown(TapDownDetails details) {
+    setState(() => isPressed = true);
+  }
 
-    return CupertinoButton(
+  void _onTapUp(TapUpDetails details) {
+    setState(() => isPressed = false);
+    widget.onPressed?.call();
+  }
+
+  void _onTapCancel() {
+    setState(() => isPressed = false);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final Widget child = CupertinoButton(
+      color: isPressed
+        ? _kToolbarPressedColor.resolveFrom(context)
+        : const Color(0x00000000),
       borderRadius: null,
-      color: _kToolbarBackgroundColor,
-      disabledColor: _kToolbarBackgroundColor,
-      onPressed: onPressed,
+      disabledColor: const Color(0x00000000),
+      onPressed: widget.onPressed,
       padding: _kToolbarButtonPadding,
-      pressedOpacity: onPressed == null ? 1.0 : 0.7,
-      child: child,
+      // There's no foreground fade on iOS toolbar anymore, it just darkens the
```

Comment:
`CupertinoColors.inactiveGray` is a `CupertinoDynamicColor` so I thought it would stay OK in dark mode too, but I tested anyway :)

![Screenshot_20230606_215359](https://github.com/flutter/flutter/assets/12024080/9569f838-c60d-43e2-a598-dabbb9958438)